src/util/worker/mod.rs

```diff
@@ -158,18 +144,18 @@ impl<T: Display> Scheduler<T> {
     /// If the worker is stopped or number pending tasks exceeds capacity, an error will return.
     pub fn schedule(&self, task: T) -> Result<(), ScheduleError<T>> {
         debug!("scheduling task {}", task);
-        match self.sender {
-            TaskSender::Unbounded(ref tx) => if let Err(SendError(Some(t))) = tx.send(Some(task)) {
+        // whether sender is bounded or not
+        if self.sender.capacity().is_none() {
```

Comment:
Why not always use `try_send`?

Reply:
Maybe we should keep the logic the same as before.

Reply:
How is it different from before?

Reply:
Previously we use `try_send` only when `sender` is bounded.

Reply:
So what's the motivation behind that?

Reply:
I think the error is different, when it is bounded, there are two errors, we should distinguish them. @hicqu